### DISCUSSION TITLE
Removing "is-plain-object" dependency

### DISCUSSION THREAD

**Discussion by fatihaziz** [RebeccaStevens/deepmerge-ts#101] [Ideas]
we could replace it with this:
```
// src/utils.ts
const isPlainObject = (obj: unknown) => {
  /* eslint-disable no-prototype-builtins, unicorn/consistent-function-scoping */
  const isDefinedObject = (obj: unknown): obj is object =>
    typeof obj === "object" &&
    obj !== null &&
    Object.prototype.toString.call(obj) === "[object Object]";

   if(obj === null) return true;
   if (!isDefinedObject(obj)) return false;

   const { constructor } = obj;
   if (typeof constructor === "undefined") return true;

   const { prototype } = constructor;
   return isDefinedObject(prototype) && prototype.hasOwnProperty("isPrototypeOf");
  /* eslint-enable no-prototype-builtins, unicorn/consistent-function-scoping */
};

// .... rest of the codes
```
its work the same I think.
here the code sandbox with some test: https://stackblitz.com/edit/typescript-jjzlnc

its not purely my idea but I took some references from another sources, also from the is-plain-object itself, 
of course we want the same result,

so this library will become zero-dependency, 
but I know this not the goal to make the library zero dependency, 
but if we can minimize the need to depend with another library, it will be great right?

**Comment by RebeccaStevens** [comment]
Will be removed in #135 as changes to the logic need to be made.
